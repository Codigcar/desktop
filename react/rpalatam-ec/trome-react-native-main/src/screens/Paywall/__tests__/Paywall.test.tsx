import { act, render } from '@testing-utils/library'
import React from 'react'

import PaywallScreen from '../Paywall'

const postmessage = (data: string | { [key: string]: string }) => ({
  nativeEvent: { data },
})

const URL = 'https://elcomercio.pe/suscripcionesdigitales'
const SCRIPT = 'console.log("inject")'
jest.mock('@react-navigation/native', () => ({
  useNavigation: jest.fn(() => ({
    goBack: jest.fn(),
  })),
  useRoute: jest.fn(() => ({
    params: {
      url: URL,
      injectedJavaScriptBeforeContentLoaded: SCRIPT,
    },
  })),
}))

const mockVerifyFn = jest.fn()
jest.mock('../../../context/auth', () => ({
  useAuth: jest.fn(() => ({
    verifySubscription: mockVerifyFn,
  })),
}))

describe('Paywall', () => {
  it('Mount screen', async () => {
    const { getByTestId, toJSON } = render(<PaywallScreen />)

    await act(async () => {
      expect(toJSON()).toMatchSnapshot()
    })

    const webview = getByTestId('paywall-screen')
    expect(webview.props.source.uri).toBe(URL)
    expect(webview.props.injectedJavaScriptBeforeContentLoaded).toBe(SCRIPT)

    // Send postmessages
    const data = { type: 'Test' }
    webview.props.onMessage(postmessage(data))
    webview.props.onMessage(postmessage(JSON.stringify(data)))

    // Verify Subscripcion
    webview.props.onMessage(postmessage('successful_purchase'))
    expect(mockVerifyFn).toBeCalled()

    // Go Back
    webview.props.onMessage(postmessage('navigation.GO_BACK'))
  })
})
