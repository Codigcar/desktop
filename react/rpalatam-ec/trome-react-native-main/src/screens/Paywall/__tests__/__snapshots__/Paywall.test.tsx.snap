// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paywall Mount screen 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "display": "flex",
          "flex": 1,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "overflow": "hidden",
          },
          undefined,
        ]
      }
    >
      <RNCWebView
        cacheEnabled={true}
        incognito={true}
        injectedJavaScriptBeforeContentLoaded="console.log(\\"inject\\")"
        injectedJavaScriptBeforeContentLoadedForMainFrameOnly={true}
        injectedJavaScriptForMainFrameOnly={true}
        javaScriptEnabled={true}
        messagingEnabled={true}
        onContentProcessDidTerminate={[Function]}
        onHttpError={[Function]}
        onLoadingError={[Function]}
        onLoadingFinish={[Function]}
        onLoadingProgress={[Function]}
        onLoadingStart={[Function]}
        onMessage={[Function]}
        onShouldStartLoadWithRequest={[Function]}
        source={
          Object {
            "uri": "https://elcomercio.pe/suscripcionesdigitales",
          }
        }
        style={
          Array [
            Object {
              "flex": 1,
              "overflow": "hidden",
            },
            Object {
              "backgroundColor": "#ffffff",
            },
            Object {
              "flex": 1,
            },
          ]
        }
        testID="paywall-screen"
        textInteractionEnabled={true}
        useSharedProcessPool={true}
      />
      <ActivityIndicator
        color="#0089ff"
        size="large"
        style={
          Object {
            "flex": 1,
            "justifyContent": "flex-start",
          }
        }
      />
    </View>
  </View>
</RNCSafeAreaProvider>
`;
